src/main/scala/io/iohk/ethereum/blockchain/sync/regular/BlockImporter.scala

```diff
@@ -227,12 +226,9 @@ class BlockImporter(
 
   private def broadcastNewBlocks(blocks: List[NewBlock]): Unit = broadcaster ! BroadcastBlocks(blocks)
 
-  private def updateTxAndOmmerPools(blocksAdded: Seq[Block], blocksRemoved: Seq[Block]): Unit = {
-    blocksRemoved.headOption.foreach(block => ommersPool ! AddOmmers(block.header))
+  private def updateTxPool(blocksAdded: Seq[Block], blocksRemoved: Seq[Block]): Unit = {
```

Comment:
I don't like much the middle ground of keeping the pool and ignoring any problems when using it on the miner, but only updating the pool it half of the times, should we:
1. Remove the pool entirely and have no support with mining blocks with ommers
2. Keep the pool updates as on develop and use the patch for producing blocks without ommers for handling the case that it has any invalid one